### PR TITLE
Example for payment gateway

### PR DESCRIPTION
**Pull request by leonthaibao**
This example is based on the Dynamic QR Code example, so you should focus on these new (updated) files:
- pages/api/create_order/index.js
- pages/api/get_list_banks/index.js
- pages/callback/index.js
- pages/cart/index.js

checkout-demo

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,36 @@
+{
+  "name": "quickstart-nextjs-dynamic-qrcode",
```
Comment (longxice) [inline package.json]: The project name should be "... checkout"

File: pages/_app.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import { Header, Footer } from '@/components/index';
+import { Toaster } from 'react-hot-toast';
+import "./css/app.css"
+
+function MyApp({ Component, pageProps }) {
+  return (
+    <>
+      <Head>
+        <title>
+          ZaloPay Dynamic QR payment integration demo
```
Comment (longxice) [inline pages/_app.js]: should be related to checkout

File: pages/api/callback/index.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    return;
+  }
+  try {
+    let result = {};
+    try {
+      let dataStr = req.body.data;
+      let reqMac = req.body.mac;
+
+      let mac = CryptoJS.HmacSHA256(dataStr, configZLP.key2).toString();
+      console.log("mac =", mac);
```
Comment (longxice) [inline pages/api/callback/index.js]: this log should be removed?

File: pages/api/query_status/index.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        'Content-Type': 'application/x-www-form-urlencoded'
+      },
+      data: qs.stringify(postData)
+    };
+
+    axios(postConfig)
+    .then(function (response) {
+      res.status(200).json(response.data);
+    })
+    .catch(function (error) {
```
Comment (longxice) [inline pages/api/query_status/index.js:37]: two catch blocks are same function?

Comment (leonthaibao) [reply pages/api/query_status/index.js:37]: There are two catch blocks in this function
The first one is the try-catch block. If there is any error in the try-block code, the catch will handle the error
The second one is to handle the error of promise when calling the API with Axios only.

**Comment by anhldbk** [comment]
@qtdan PTAL

File: README.md [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+**Note**
+
+This example doesn't include the callback when run at localhost, see how to integration callback in next section
+
+## Troubleshooting
+
+### Callback testing
+
+Callback (a.k.a Webhook) is a mechanism to deliver Instant Payment Notification asynchronously for Merchant Server receives payment's results from ZaloPay Server, only when ZaloPay has received money from user successfully, and it is important to test them during the setup of your integration.
+You can find more information about callbacks in [here](https://docs.zalopay.vn/en/v2/general/overview.html#callback).
```
Comment (qtdan) [inline README.md]: Can we link to the latest version if capable https://beta-docs.zalopay.vn/

File: README.md [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+    - [Making your server reachable](#making-your-server-reachable)
+    - [Callback setup](#callback-setup)
+  - [Author(s)](#authors)
+  - [Acknowledgement](#acknowledgement)
+
+
+
+## Overview
+
+This repository includes examples of integrations for online payments with ZaloPay [Payment Gateway](https://docs.zalopay.vn/v2/payments/gateway/overview.html). Within this demo app, you'll find a simplified version of an e-commerce website, complete with commented code to highlight key features and concepts of ZaloPay's API.
```
Comment (anhldbk) [inline README.md]: @leonthaibao anh nghĩ nên bỏ link cũ (https://docs.zalopay.vn/v2/payments/gateway/overview.html). Nếu không, hôm sau beta-docs lên thay thế cái cũ thì em sẽ phải làm lại merge request này.

File: README.md [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+
+## Requirements
+
+Node.js 16+
+
+## Installation
+
+1. Clone this repo:
+
+```
```
Comment (anhldbk) [inline README.md]: @leonthaibao lệnh bash -> anh nghĩ em vẫn nên viết thêm `sh`

File: public/images/bank-abb.svg
Comment (qtdan) [inline public/images/bank-abb.svg:1]: why do we need all these bank logos?
Do you mean that the get bank api doesn't give your the logo url :D.

File: pages/api/create_order/index.js [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+    }
+    if (paymentType === "ATM") {
+      embed_data.bankgroup = "ATM"
+    }
+
+    if (paymentType !== "ALL") {
+      bankCode = paymentType
+    }
+
+    const order = {
```
Comment (qtdan) [inline pages/api/create_order/index.js:38]: we still agree to hard code those order information?

File: public/images/bank-abb.svg
Comment (leonthaibao) [reply public/images/bank-abb.svg:1]: That's right. Here is an example of the response of a bank:
```
            {
                "bankcode": "PVCB",
                "name": "PVCombank",
                "displayorder": 1,
                "pmcid": 39,
                "minamount": 10000,
                "maxamount": 250000000
            },
```
And in the previous doc, we publish some HTML, and CSS code with the bank logos. But it's outdated
In the case of Tiki, I see that they use their logos which are different from ours.
So with the show the list of the bank on the merchant site approach, the merchant has to handle the logos by themselves

File: pages/api/create_order/index.js (hunk 8 above)
Comment (leonthaibao) [reply pages/api/create_order/index.js:38]: I will adjust the BE API to handle the real amount number

**Comment by leonthaibao** [comment]
@anhldbk  @qtdan All comments have been updated, pls re-review

**Review by qtdan — APPROVED** [review]
LGTM, thanks

**Pull request merged**